File: Peter/누적합/BOJ_11659.js
// 문제: https://www.acmicpc.net/problem/11659
const fs = require('fs');
// let input = fs.readFileSync('/dev/stdin').toString().split('\n');
const input = fs.readFileSync('./input.txt').toString().split('\r\n');

const [firstLine, numberString, ...ranges] = input;
const [count, length] = firstLine.split(' ').map(Number);
const numbers = numberString.split(' ').map(Number);

for (let i = 1; i < count; i++) {
  numbers[i] = numbers[i - 1] + numbers[i];
}
numbers.unshift(0);

for (let i = 0; i < length; i++) {
  const range = ranges[i].split(' ');
  const [start, end] = range.map(Number);
  console.log(numbers[end] - numbers[start - 1]);
}

File: Peter/누적합/BOJ_11659.py
import sys
sys.stdin = open('input.txt', 'r')

count, length = map(int, sys.stdin.readline().split())
numbers = sys.stdin.readline().split()

# 앞에 0을 넣어주면 배열 index 벗어날 일이 없어서 편함
numbers.insert(0, 0)

# 누적합 배열 구함
for i in range(count):
  numbers[i + 1] = int(numbers[i + 1]) + int(numbers[i])

# 끝 값 - (시작 - 1) 값 = 범위 누적합
for i in range(length):
  start, end = sys.stdin.readline().split()
  print(numbers[int(end)] - numbers[int(start) - 1])
